include/libtwofloat/arithmetics/double-word-arithmetic.hpp

```diff
@@ -233,5 +233,628 @@ inline two<T> div(const two<T> &x, const two<T> &y) {
+namespace qd {
+
+  // Reference: QD / inline.h
+  /* Computes the nearest integer to input. */
+  template <typename T>
+  inline T nint(T input) {
+    T pointfive;
+
+    if constexpr (std::is_same_v<T, float>) {
+      pointfive = 0.5f;
+    } else if constexpr (std::is_same_v<T, double>) {
+      pointfive = 0.5;
+    } else {
+      std::cerr << "Other types not supported\n";
+    }
+
+    if(input = std::floor(input)) {
+      return input;
+    }
+    return std::floor(input + pointfive);
+  }
+
+  // TODO: already implemented in twofloat?
+  // Reference: QD / inline.h
+  /* Computes fl(a+b) and err(a+b). Assumes |a| >= |b|. */
+  template <typename T>
+  inline T quick_two_sum(T a, T b, T &err) {
+    T s = a + b;
+    err = b - (s - a);
+    return s;
+  }
+
+  // TODO: make sure it has not been already implemented in twofloat
+  // Reference: QD / inline.h
+  /* Computes high word and low word of input */
+  template <typename T>
+  inline void split(T input, T &hi, T &lo) {
+    T temp;
+
+    T local_qd_splitter;
+    T local_qd_split_thresh;
+    T local_split_factor;
+    T local_split_factor_2;
+    if constexpr (std::is_same_v<T, float>) {
+      local_qd_splitter = fp32_qd_splitter;
+      local_qd_split_thresh = fp32_qd_split_thresh;
+      local_split_factor = fp32_split_factor;
+      local_split_factor_2 = fp32_split_factor_2;
+    } else if constexpr (std::is_same_v<T, double>) {
+      local_qd_splitter = fp64_qd_splitter;
+      local_qd_split_thresh = fp64_qd_split_thresh;
+      local_split_factor = fp64_split_factor;
+      local_split_factor_2 = fp64_split_factor_2;
+    } else {
+      std::cerr << "Other types not supported\n";
+    }
+
+    if(input > local_qd_split_thresh || input < -local_qd_split_thresh) {
+      input *= local_split_factor;
+      temp = local_qd_splitter * input;
+      hi = temp - (temp - input);
+      lo = input - hi;
+      hi *= local_split_factor_2;
+      lo *= local_split_factor_2;
+    } else {
+      temp = local_qd_splitter * input;
+      hi = temp - (temp - input);
+      lo = input - hi;
+    }
+  }
+
+  // Reference: QD / inline.h
+  /* Computes fl(input * input) and err(input * input) */
+  template <typename T>
+  inline T two_sqr(T input, T &err) {
+
+    T twopointzero;
+    if constexpr (std::is_same_v<T, float>) {
+      twopointzero = 2.0f;
+    } else if constexpr (std::is_same_v<T, double>) {
+      twopointzero = 2.0;
+    } else {
+      std::cerr << "Other types not supported\n";
+    }
+
+    T hi, lo;
+    T q = input * input;
+    qd::split(input, hi, lo);
+    err = ((hi * hi - q) + twopointzero * hi * lo) + lo * lo;
+    return q;
+  }
+
+  // Reference QD / inline.h
+  // TODO: make sure this has not been already implemented
+  template <typename T>
+  inline T two_sum(T a, T b, T &err) {
+    T s = a + b;
+    T bb = s - a;
+    err = (a - (s - bb)) - (b - bb);
+    return s;
+  }
+
+} // End namespace qd
+
+namespace dd_real {
```

Comment:
Whats this namespace for?

Reply:
Namespace `dd_real` has been removed in [2421180](https://github.com/esa-tu-darmstadt/twofloat/pull/1/commits/242118023c7c5b21ca0d29ab4f3db7e1e0e5d517).

This is related to https://github.com/esa-tu-darmstadt/twofloat/pull/1#discussion_r1482063167.